auto3dseg/algorithm_templates/segresnet/configs/hyper_parameters.yaml

```diff
@@ -1,63 +1,78 @@
-# hyper-parameters
-amp: true
-bundle_root: null
-ckpt_path: "$@bundle_root + '/model_fold' + str(@fold)"
-data_file_base_dir: null
-data_list_file_path: null
+modality: mri                               # main image modality, must be one of mri, ct, pet
+data_file_base_dir: null                    # location of the dataset
+data_list_file_path: null                   # location of the file with a list of files image/label in the dataset
+
+input_channels: null
+output_classes: null
+
 class_names: null
 class_index: null
+
+quick: false
+
+roi_size: [128, 128, 128]
+crop_mode: ratio
+normalize_mode: meanstd
+
+amp: true
+bundle_root: null
+ckpt_path: "$@bundle_root + '/model'"
+ckpt_save: true
+cache_rate: null
+
 determ: false
 fold: 0
-input_channels: null
 learning_rate: 2.0e-4
-num_images_per_batch: 2
-num_epochs: 1250
-num_epochs_per_validation: 1
-num_patches_per_image: 1
-num_sw_batch_size: 2
-output_classes: null
-overlap_ratio: 0.625
-patch_size: null
-patch_size_valid: null
-softmax: true
+num_images_per_batch: 1
+num_epochs: 300
+warmup_epochs: 3
+sigmoid: false
 resample: false
+resample_resolution: [1, 1, 1]
 intensity_bounds: null
 
-# training
+num_epochs_per_validation: 1
+num_epochs_per_saving: 1
+num_workers: 4
+
 loss:
-  _target_: loss.DiceCELoss2
+  _target_: DiceCELoss
   include_background: true
-  sigmoid: $not @softmax
-  smooth_dr: 1.0e-05
-  smooth_nr: 0
-  softmax: $@softmax
   squared_pred: true
-  to_onehot_y: $@softmax
+  smooth_nr: 0
+  smooth_dr: 1.0e-05
+  softmax: $not @sigmoid
+  sigmoid: $@sigmoid
+  to_onehot_y: $not @sigmoid
+
 optimizer:
   _target_: torch.optim.AdamW
   lr: "@learning_rate"
-  weight_decay: 1.0e-05
-lr_scheduler:
-  _target_: flow_cosinelr.CosineAnnealingLRWithWarmup
-  optimizer: "$@optimizer"
-  warmup_epochs: 20
-  warmup_multiplier: 0.1
-  T_max: "@num_epochs"
-
-# fine-tuning
+  weight_decay: 1.e-5
+
+network:
+  _target_: SegResNetDS
```

Comment:
Hi @myron, I noticed that you have changed the loss function, removed lr_scheduler, and used a new version SegResNet. Not sure if it makes more sense to change the name of this whole template to segresnetds or just add a new template to keep the original segresnet if there are users who still want to use the original version. What do you think? 
In addition to that, perhaps we need to update the results in the tutorial repo. 
https://github.com/Project-MONAI/tutorials/tree/main/auto3dseg/tasks/msd
Thanks!

Reply:
thank you for the comment. the loss function is the same (it's just a loss with deep supervision), and the network is the same (but with a deep supervision support).  I don't think we need another template, it'll be difficult to maintain.  and we provide a versioned tar.gz release url to download the algo templates (if anyone wants to use the older variant)

regarding updates to the tutorials repo,  we can can consider that after this is merged. thank you

Reply:
Hi @myron, thanks for your explanation, I know the network is the same then maybe we could change this `dsdepth` to 1? Users may not be able to notice that this new version of the template has added deep supervision support.
https://github.com/Project-MONAI/research-contributions/blob/fde97625e100e1d44d029fcc98186efd1a2a190c/auto3dseg/algorithm_templates/segresnet/configs/hyper_parameters.yaml#L60

And why do you need to remove the learning rate scheduler？ For better performance or anything else? Could you please help explain it? 
I noticed that you have simplified the config, putting all the configs in the same file. This does look clearer and simpler. I remember we had a similar discussion before about why we need to separate the config into different files, it seems to prevent overwriting problems caused by duplicate parameters. Maybe there is no such problem in this template, I'm not sure if we could keep both ways of writing the config. Could you please help confirm it? @dongyang0122 @mingxin-zheng @wyli, Thanks!

Reply:
if you notice that some options such as dsdepth have changed, it's because they are updated to be the proper  values, that we should have used from the beginning, and here we're fixing them. 

regarding the learning rate scheduler. this particular "algorithm template" is a full workflow , it's not just a network/model with arbitrary training options. and in this workflow we use a particular learning rate scheduler.  Same for many other components used during training. We find a balance, by exposing a large set of parameters, but not every single class. 

if the config file is very long, it is split into several files purely for visual convenience (they are all merged during loading). Here, there is no need to split into sub configs. thanks.

Reply:
I saw a style change in the way to organize the code. For example, transforms are taken out from the bundle configs. I think this is the new direction after discussions and I am completely fine with it. Thanks!